File: 2021년/9월/0923 경로 찾기.cpp
#include <iostream>
#include <algorithm>
using namespace std;

int n;
int graph[100][100];
int INF = 1000000;

int main()
{
	scanf("%d", &n);
	for(int i=0;i<n;i++)
	{
		for(int j=0;j<n;j++)
		{
			int x;
			scanf("%d", &x);
			if(x==0) graph[i][j] = INF;
			else graph[i][j] = x;
		}
	}
	
	for(int k=0;k<n;k++)
	{
		for(int i=0;i<n;i++)
		{
			for(int j=0;j<n;j++)
			{
				graph[i][j] = min(graph[i][j], graph[k][j]+graph[i][k]);
			}
		}
	}
	
	for(int i=0;i<n;i++)
	{
		for(int j=0;j<n;j++)
		{
			if(graph[i][j] == INF) printf("%d ", 0);
			else printf("%d ", 1);
		} 
		printf("\n");
	}
}

File: 2021년/9월/0923 경로 찾기.py
# 백준_11403_최단경로_플로이드-와샬_실버 1

import sys
input = sys.stdin.readline
INF = sys.maxsize

n = int(input())
graph = []

for i in range(n):
    lst = list(map(int, input().split()))
    for j in range(n):
        if lst[j] == 0:
            lst[j] = INF
    graph.append(lst)

for k in range(n):
    for i in range(n):
        for j in range(n):
            graph[i][j] = min(graph[i][j], graph[k][j]+graph[i][k])

for i in range(n):
    for j in range(n):
        if graph[i][j] == INF:
            print(0, end=' ')
        else:
            print(1, end=' ')
    print()
